# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/AccEvent.h
# Commit: 786805a2cdcc
# Full Hash: 786805a2cdcc4cdab632a955dace417062d0087e
# Author: James Teh <jteh@mozilla.com>
# Date: 2023-11-30 04:12:04
# Regressor Bug: 1862802
# File Overlap Count: 1
# Description:
#   Bug 1862802 part 5: Remove virtual cursor change event. r=eeejay
#   
#   Depends on D192645
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D192646
# ==============================================================================

diff -r effe12f2120a -r 786805a2cdcc accessible/base/AccEvent.h
--- a/accessible/base/AccEvent.h	Tue Nov 28 22:03:20 2023 +0000
+++ b/accessible/base/AccEvent.h	Tue Nov 28 22:03:21 2023 +0000
@@ -102,7 +102,6 @@
     eCaretMoveEvent,
     eTextSelChangeEvent,
     eSelectionChangeEvent,
-    eVirtualCursorChangeEvent,
     eObjectAttrChangedEvent,
     eScrollingEvent,
     eAnnouncementEvent,
@@ -442,35 +441,6 @@
 };
 
 /**
- * Accessible virtual cursor change event.
- */
-class AccVCChangeEvent : public AccEvent {
- public:
-  AccVCChangeEvent(LocalAccessible* aAccessible,
-                   LocalAccessible* aOldAccessible,
-                   LocalAccessible* aNewAccessible, int16_t aReason,
-                   EIsFromUserInput aIsFromUserInput = eFromUserInput);
-
-  virtual ~AccVCChangeEvent() {}
-
-  // AccEvent
-  static const EventGroup kEventGroup = eVirtualCursorChangeEvent;
-  virtual unsigned int GetEventGroups() const override {
-    return AccEvent::GetEventGroups() | (1U << eVirtualCursorChangeEvent);
-  }
-
-  // AccVCChangeEvent
-  LocalAccessible* OldAccessible() const { return mOldAccessible; }
-  LocalAccessible* NewAccessible() const { return mNewAccessible; }
-  int32_t Reason() const { return mReason; }
-
- private:
-  RefPtr<LocalAccessible> mOldAccessible;
-  RefPtr<LocalAccessible> mNewAccessible;
-  int16_t mReason;
-};
-
-/**
  * Accessible object attribute changed event.
  */
 class AccObjectAttrChangedEvent : public AccEvent {